src/util/hass-translation.ts

```diff
@@ -1,58 +1,63 @@
 import { translationMetadata } from "../resources/translations-metadata";
+import { fetchFrontendUserData } from "../data/frontend-user-data";
+import { HomeAssistant } from "../types";
 
-export function getActiveTranslation() {
+/**
+ * Search for a matching translation from most specific to general
+ */
+function findAvilableLanguage(language: string) {
   // Perform case-insenstive comparison since browser isn't required to
   // report languages with specific cases.
-  const lookup = {};
-  /* eslint-disable no-undef */
+  const avilableLanguages = {
+    "zh-cn": "zh-Hans",
+    "zh-sg": "zh-Hans",
+    "zh-my": "zh-Hans",
+    "zh-tw": "zh-Hant",
+    "zh-hk": "zh-Hant",
+    "zh-mo": "zh-Hant",
+  };
   Object.keys(translationMetadata.translations).forEach((tr) => {
-    lookup[tr.toLowerCase()] = tr;
+    avilableLanguages[tr.toLowerCase()] = tr;
   });
 
-  // Search for a matching translation from most specific to general
-  function languageGetTranslation(language) {
-    const lang = language.toLowerCase();
+  return avilableLanguages[language.toLowerCase()];
```

Comment:
This seems inefficient. We don't need to convert all languages to lowercase to find the one matching.

```ts
const langLower = language.toLowerCase();
for (const lang in Object.keys(translationMetadata.translations)) {
  if (lang.toLowerCase() === langLower) {
    return lang;
  }
}
```

Reply:
Done